crates/router/src/connector/nexinets/transformers.rs

```diff
@@ -54,61 +130,236 @@ pub struct NexinetsAuthType {
+            api_models::enums::BankNames::Bunq => Ok(Self::Bunq),
+            api_models::enums::BankNames::Ing => Ok(Self::Ing),
+            api_models::enums::BankNames::Knab => Ok(Self::Knab),
+            api_models::enums::BankNames::Rabobank => Ok(Self::Rabobank),
+            api_models::enums::BankNames::Regiobank => Ok(Self::Regiobank),
+            api_models::enums::BankNames::SnsBank => Ok(Self::SnsBank),
+            api_models::enums::BankNames::TriodosBank => Ok(Self::TriodosBank),
+            api_models::enums::BankNames::VanLanschot => Ok(Self::VanLanschot),
+            _ => Err(errors::ConnectorError::FlowNotSupported {
+                flow: bank.to_string(),
+                connector: "Nexinets".to_string(),
+            }
+            .into()),
         }
     }
 }
 
-#[derive(Default, Debug, Clone, Serialize, Deserialize, PartialEq)]
-pub struct NexinetsPaymentsResponse {
-    status: NexinetsPaymentStatus,
-    id: String,
+#[derive(Debug, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct NexinetsPreAuthOrDebitResponse {
+    order_id: String,
+    transaction_type: NexinetsTransactionType,
+    transactions: Vec<NexinetsTransaction>,
+    redirect_url: Option<Url>,
+}
+
+#[derive(Debug, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct NexinetsTransaction {
+    pub transaction_id: String,
+    #[serde(rename = "type")]
+    pub transaction_type: NexinetsTransactionType,
+    pub currency: String,
+    pub status: NexinetsPaymentStatus,
+}
+
+#[derive(Default, Debug, Serialize, Deserialize)]
+#[serde(rename_all = "SCREAMING_SNAKE_CASE")]
+pub enum NexinetsTransactionType {
+    #[default]
+    Preauth,
+    Debit,
+    Capture,
+    Cancel,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct NexinetsPaymentsMetadata {
+    pub transaction_id: Option<String>,
 }
 
 impl<F, T>
-    TryFrom<types::ResponseRouterData<F, NexinetsPaymentsResponse, T, types::PaymentsResponseData>>
-    for types::RouterData<F, T, types::PaymentsResponseData>
+    TryFrom<
+        types::ResponseRouterData<
+            F,
+            NexinetsPreAuthOrDebitResponse,
+            T,
+            types::PaymentsResponseData,
+        >,
+    > for types::RouterData<F, T, types::PaymentsResponseData>
 {
     type Error = error_stack::Report<errors::ConnectorError>;
     fn try_from(
         item: types::ResponseRouterData<
             F,
-            NexinetsPaymentsResponse,
+            NexinetsPreAuthOrDebitResponse,
             T,
             types::PaymentsResponseData,
         >,
     ) -> Result<Self, Self::Error> {
+        let transaction = match item.response.transactions.first() {
+            Some(order) => order,
+            _ => Err(errors::ConnectorError::ResponseHandlingFailed)?,
+        };
+        let connector_metadata = serde_json::to_value(NexinetsPaymentsMetadata {
+            transaction_id: Some(transaction.transaction_id.clone()),
```

Comment:
if transaction_id is used for cancel, capture and sync then this should be connector_id right? why are are passing order_id as connector_id?

Reply:
In the response, we get two connector transaction IDs:
-order_id => remains the same throughout the payment_attempt; this is stored in connector_transaction_id.
-transaction_id => changes for every subsequent action (cancel, refund, capture, partial refund); this is stored in connector_meta_data.
-Both order_id and transaction_id are used for further actions (Psync, Cancel, Rsync, Refund, Capture), which should be passed in the URL.